babel/src/main/codegen/includes/parserImpls.ftl

```diff
@@ -505,13 +507,56 @@ SqlCreate SqlCreateTable(Span s, boolean replace) :
     |
         { query = null; }
     )
+    (
+       index = SqlCreateTableIndex(s)
+       {
+           if(index instanceof SqlPrimaryIndex){
+               primaryIndex = (SqlPrimaryIndex) index;
```

Comment:
Is this casting necessary? Would it work to instead have primary index be of type SqlIndex (in the SqlCreateTable class) and then just pass in the index as is to the constructor? Unparsing should be unaffected.

Reply:
The idea is that we can eventually extend the SqlIndex class to include other types of indices, not just primary indices. However, we wouldn't want any of those other SqlIndex subclasses to be passed in to the primaryIndex field of SqlCreateTable, since that should only ever be a SqlPrimaryIndex.

Reply:
Ah alright, makes sense.